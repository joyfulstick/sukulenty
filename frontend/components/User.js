import PropTypes from 'prop-types'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'

const CURRENT_USER_QUERY = gql`
  query {
    me {
      id
      email
      name
      permissions
      orders {
        id
      }
      cart {
        id
        quantity
        item {
          id
          title
          price
          image
        }
      }
    }
  }
`
const User = props => (
  <Query query={CURRENT_USER_QUERY} {...props}>
    {payload => props.children(payload)}
  </Query>
)

User.propTypes = {
  children: PropTypes.func.isRequired,
}

export { User as default, CURRENT_USER_QUERY }
