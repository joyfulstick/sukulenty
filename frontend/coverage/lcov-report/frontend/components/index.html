<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for frontend/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> frontend/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">47.77% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>75/157</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>21/50</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">32.79% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>20/61</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">49.31% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>71/144</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="AddToCart.js"><a href="AddToCart.js.html">AddToCart.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Cart.js"><a href="Cart.js.html">Cart.js</a></td>
	<td data-value="30.77" class="pic low"><div class="chart"><div class="cover-fill" style="width: 30%;"></div><div class="cover-empty" style="width:70%;"></div></div></td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="12" class="abs low">4/12</td>
	</tr>

<tr>
	<td class="file high" data-value="CartCount.js"><a href="CartCount.js.html">CartCount.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file low" data-value="CartItem.js"><a href="CartItem.js.html">CartItem.js</a></td>
	<td data-value="42.86" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	</tr>

<tr>
	<td class="file low" data-value="Checkout.js"><a href="Checkout.js.html">Checkout.js</a></td>
	<td data-value="9.09" class="pic low"><div class="chart"><div class="cover-fill" style="width: 9%;"></div><div class="cover-empty" style="width:91%;"></div></div></td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="10" class="pct low">10%</td>
	<td data-value="10" class="abs low">1/10</td>
	</tr>

<tr>
	<td class="file low" data-value="DeleteItem.js"><a href="DeleteItem.js.html">DeleteItem.js</a></td>
	<td data-value="12.5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 12%;"></div><div class="cover-empty" style="width:88%;"></div></div></td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="16" class="abs low">2/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="13.33" class="pct low">13.33%</td>
	<td data-value="15" class="abs low">2/15</td>
	</tr>

<tr>
	<td class="file high" data-value="ErrorMessage.js"><a href="ErrorMessage.js.html">ErrorMessage.js</a></td>
	<td data-value="81.82" class="pic high"><div class="chart"><div class="cover-fill" style="width: 81%;"></div><div class="cover-empty" style="width:19%;"></div></div></td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	</tr>

<tr>
	<td class="file high" data-value="Guard.js"><a href="Guard.js.html">Guard.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="Item.js"><a href="Item.js.html">Item.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file low" data-value="Items.js"><a href="Items.js.html">Items.js</a></td>
	<td data-value="30.77" class="pic low"><div class="chart"><div class="cover-fill" style="width: 30%;"></div><div class="cover-empty" style="width:70%;"></div></div></td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	</tr>

<tr>
	<td class="file high" data-value="Nav.js"><a href="Nav.js.html">Nav.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file low" data-value="Pagination.js"><a href="Pagination.js.html">Pagination.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="12" class="abs low">2/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="11" class="abs low">2/11</td>
	</tr>

<tr>
	<td class="file low" data-value="RemoveFromCart.js"><a href="RemoveFromCart.js.html">RemoveFromCart.js</a></td>
	<td data-value="15.38" class="pic low"><div class="chart"><div class="cover-fill" style="width: 15%;"></div><div class="cover-empty" style="width:85%;"></div></div></td>
	<td data-value="15.38" class="pct low">15.38%</td>
	<td data-value="13" class="abs low">2/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="11" class="abs low">2/11</td>
	</tr>

<tr>
	<td class="file low" data-value="Signin.js"><a href="Signin.js.html">Signin.js</a></td>
	<td data-value="36.36" class="pic low"><div class="chart"><div class="cover-fill" style="width: 36%;"></div><div class="cover-empty" style="width:64%;"></div></div></td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	</tr>

<tr>
	<td class="file high" data-value="Signout.js"><a href="Signout.js.html">Signout.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="SingleItem.js"><a href="SingleItem.js.html">SingleItem.js</a></td>
	<td data-value="92.31" class="pic high"><div class="chart"><div class="cover-fill" style="width: 92%;"></div><div class="cover-empty" style="width:8%;"></div></div></td>
	<td data-value="92.31" class="pct high">92.31%</td>
	<td data-value="13" class="abs high">12/13</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="11" class="abs high">10/11</td>
	</tr>

<tr>
	<td class="file high" data-value="User.js"><a href="User.js.html">User.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 01 2018 23:56:19 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
