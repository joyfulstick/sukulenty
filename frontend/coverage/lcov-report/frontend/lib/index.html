<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> frontend/lib
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">52.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>21/40</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">14.29% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.58% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">51.61% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/31</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="calcTotalItems.js"><a href="calcTotalItems.js.html">calcTotalItems.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file low" data-value="calcTotalPrice.js"><a href="calcTotalPrice.js.html">calcTotalPrice.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	</tr>

<tr>
	<td class="file medium" data-value="cloudinaryUtils.js"><a href="cloudinaryUtils.js.html">cloudinaryUtils.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file high" data-value="formatMoney.js"><a href="formatMoney.js.html">formatMoney.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file low" data-value="polishPurals.js"><a href="polishPurals.js.html">polishPurals.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="56.25" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 56%;"></div><div class="cover-empty" style="width:44%;"></div></div></td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="10" class="abs low">3/10</td>
	<td data-value="54.55" class="pct medium">54.55%</td>
	<td data-value="11" class="abs medium">6/11</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 01 2018 23:56:19 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
