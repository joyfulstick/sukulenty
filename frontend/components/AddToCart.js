import { CURRENT_USER_QUERY } from './User'
import { Mutation } from 'react-apollo'
import PropTypes from 'prop-types'
import React from 'react'
import gql from 'graphql-tag'

const ADD_TO_CART_MUTATION = gql`
  mutation ADD_TO_CART_MUTATION($id: ID!) {
    addToCart(id: $id) {
      id
      quantity
    }
  }
`

class AddToCart extends React.Component {
  static propTypes = {
    id: PropTypes.string,
  }
  render() {
    const { id } = this.props
    return (
      <Mutation
        mutation={ADD_TO_CART_MUTATION}
        variables={{
          id,
        }}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(addToCart, { loading }) => (
          <button
            disabled={loading}
            onClick={() => addToCart(id).catch(err => alert(err.message))}
          >
            ðŸ›’ Dodaj
            {loading && 'e'} do Koszyka
          </button>
        )}
      </Mutation>
    )
  }
}

export { AddToCart as default, ADD_TO_CART_MUTATION }
