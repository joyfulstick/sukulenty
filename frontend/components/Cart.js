import { Mutation, Query } from 'react-apollo'
import CartItem from './CartItem'
import CartStyles from './styles/CartStyles'
import CloseButton from './styles/CloseButton'
import React from 'react'
import StickButton from './styles/StickButton'
import Supreme from './styles/Supreme'
import User from './User'
import { adopt } from 'react-adopt'
import calcTotalPrice from '../lib/calcTotalPrice'
import formatMoney from '../lib/formatMoney'
import gql from 'graphql-tag'

const LOCAL_STATE_QUERY = gql`
  query {
    cartOpen @client
  }
`

const TOGGLE_CART_MUTATION = gql`
  mutation {
    toggleCart @client
  }
`
/* eslint-disable */
const Composed = adopt({
  user: ({ render }) => <User>{render}</User>,
  toggleCart: ({ render }) => (
    <Mutation mutation={TOGGLE_CART_MUTATION}>{render}</Mutation>
  ),
  localState: ({ render }) => <Query query={LOCAL_STATE_QUERY}>{render}</Query>,
})
/* eslint-enable */

const Cart = () => (
  <Composed>
    {({ user, toggleCart, localState }) => {
      const { me } = user.data
      if (!me) return null
      return (
        <CartStyles open={localState.data.cartOpen}>
          <header>
            <CloseButton onClick={toggleCart} title="close">
              &times;
            </CloseButton>
            <Supreme>{me.name}</Supreme>
            <p>Masz {me.cart.length} ðŸŒµ w koszyku</p>
          </header>
          <ul>
            {me.cart.map(cartItem => (
              <CartItem key={cartItem.id} cartItem={cartItem} />
            ))}
          </ul>
          <footer>
            <p>{formatMoney(calcTotalPrice(me.cart))}</p>
            {me.cart.length && <StickButton>ZapÅ‚aÄ‡</StickButton>}
          </footer>
        </CartStyles>
      )
    }}
  </Composed>
)

export { Cart as default, LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION }
