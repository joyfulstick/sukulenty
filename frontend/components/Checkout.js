import React, { Component } from 'react'
import User, { CURRENT_USER_QUERY } from './User'
import { Mutation } from 'react-apollo'
import NProgress from 'nprogress'
import Router from 'next/router'
import StripeCheckout from 'react-stripe-checkout'
import calcTotalPrice from '../lib/calcTotalPrice'
import gql from 'graphql-tag'
import totalItems from '../lib/calcTotalItems'

const CREATE_ORDER_MUTATION = gql`
  mutation CREATE_ORDER_MUTATION($token: String!) {
    createOrder(token: $token) {
      id
      charge
      total
      items {
        id
        title
      }
    }
  }
`
class Checkout extends Component {
  onToken = async (res, createOrder) => {
    NProgress.start()
    const order = await createOrder({
      variables: {
        token: res.id,
      },
    }).catch(err => {
      alert(err.message)
    })
    Router.push({
      pathname: '/order',
      query: { id: order.data.createOrder.id },
    })
  }
  render() {
    return (
      <User>
        {({ data: { me }, loading }) => {
          if (loading) return null
          return (
            <Mutation
              mutation={CREATE_ORDER_MUTATION}
              refetchQueries={[{ query: CURRENT_USER_QUERY }]}
            >
              {createOrder => (
                <StripeCheckout
                  amount={calcTotalPrice(me.cart)}
                  name="Sukulenty"
                  description={`ZamÃ³w ${totalItems(me.cart)} ðŸŒµ`}
                  image={
                    me.cart.length && me.cart[0].item && me.cart[0].item.image
                  }
                  stripeKey="pk_test_t4mGEh7bAv5jkeCup9hmqTae"
                  currency="PLN"
                  panelLabel="ZapÅ‚aÄ‡"
                  email={me.email}
                  token={res => this.onToken(res, createOrder)}
                  allowRememberMe={false}
                >
                  {this.props.children}
                </StripeCheckout>
              )}
            </Mutation>
          )
        }}
      </User>
    )
  }
}

export { Checkout as default, CREATE_ORDER_MUTATION }
