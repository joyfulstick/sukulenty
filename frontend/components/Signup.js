import React, { Component } from 'react'
import { CURENT_USER_QUERY } from './User'
import Error from './ErrorMessage'
import Form from './styles/Form'
import { Mutation } from 'react-apollo'
import gql from 'graphql-tag'

const SIGNUP_MUTATION = gql`
  mutation SIGNUP_MUTATION(
    $email: String!
    $password: String!
    $name: String!
  ) {
    signup(email: $email, password: $password, name: $name) {
      id
      email
      name
    }
  }
`

class Signup extends Component {
  state = {
    email: '',
    name: '',
    password: '',
  }

  handleChangeValue = e => this.setState({ [e.target.name]: e.target.value })

  render() {
    const { state, handleChangeValue } = this
    return (
      <Mutation
        mutation={SIGNUP_MUTATION}
        variables={state}
        refetchQueries={[{ query: CURENT_USER_QUERY }]}
      >
        {(signup, { error, loading }) => (
          <Form
            method="post"
            onSubmit={async e => {
              e.preventDefault()
              await signup().catch(err => console.log(err)) // eslint-disable-line
              this.setState({
                email: '',
                name: '',
                password: '',
              })
            }}
          >
            <fieldset disabled={loading} aria-busy={loading}>
              <h2>Rejestracja</h2>
              <Error error={error} />
              <label>
                Email
                <input
                  type="email"
                  name="email"
                  placeholder="e-mail"
                  value={state.email}
                  onChange={handleChangeValue}
                  required
                />
              </label>
              <label>
                Imię
                <input
                  type="text"
                  name="name"
                  placeholder="Imię"
                  value={state.name}
                  onChange={handleChangeValue}
                  required
                />
              </label>
              <label>
                Hasło
                <input
                  type="password"
                  name="password"
                  placeholder="Hasło"
                  value={state.password}
                  onChange={handleChangeValue}
                  required
                />
              </label>
              <button type="submit">Zapisz się</button>
            </fieldset>
          </Form>
        )}
      </Mutation>
    )
  }
}

export default Signup
