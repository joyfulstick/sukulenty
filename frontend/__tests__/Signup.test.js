import Signup, { SIGNUP_MUTATION } from '../components/Signup'
import { ApolloConsumer } from 'react-apollo'
import { CURRENT_USER_QUERY } from '../components/User'
import { MockedProvider } from 'react-apollo/test-utils'
import Router from 'next/router'
import { fakeUser } from '../lib/testUtils'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import wait from 'waait'

Router.router = {
  push() {},
}

function type(wrapper, name, value) {
  wrapper.find(`input[name="${name}"]`).simulate('change', {
    target: { name, value },
  })
}

const me = fakeUser()
const mocks = [
  {
    request: {
      query: SIGNUP_MUTATION,
      variables: {
        name: me.name,
        email: me.email,
        password: '123',
      },
    },
    result: {
      data: {
        signup: {
          __typename: 'User',
          id: 'abc123',
          email: me.email,
          name: me.name,
        },
      },
    },
  },
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me } },
  },
]

describe('<Signup/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <Signup />
      </MockedProvider>,
    )
    expect(toJSON(wrapper.find('form'))).toMatchSnapshot()
  })

  it('calls the mutation properly', async () => {
    let apolloClient
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ApolloConsumer>
          {client => {
            apolloClient = client
            return <Signup />
          }}
        </ApolloConsumer>
      </MockedProvider>,
    )
    await wait()
    wrapper.update()
    type(wrapper, 'name', me.name)
    type(wrapper, 'email', me.email)
    type(wrapper, 'password', '123')
    wrapper.update()
    wrapper.find('form').simulate('submit')
    await wait()
    const user = await apolloClient.query({ query: CURRENT_USER_QUERY })
    expect(user.data.me).toMatchObject(me)
  })
})
