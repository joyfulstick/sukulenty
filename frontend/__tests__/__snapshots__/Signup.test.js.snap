// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> renders and matches snapshot 1`] = `
<form
  className="Form-t0rxhl-0 lpjQDY"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Rejestracja
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label>
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="e-mail"
        required={true}
        type="email"
        value=""
      />
    </label>
    <label>
      Imię
      <input
        name="name"
        onChange={[Function]}
        placeholder="Imię"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label>
      Hasło
      <input
        name="password"
        onChange={[Function]}
        placeholder="Hasło"
        required={true}
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Zapisz się
    </button>
  </fieldset>
</form>
`;
